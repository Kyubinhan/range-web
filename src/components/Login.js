import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Button, Segment } from 'semantic-ui-react';
import { Loading } from './common';
import { SSO_LOGIN_ENDPOINT, SSO_IDIR_LOGIN_ENDPOINT, SSO_BCEID_LOGIN_ENDPOINT } from '../constants/API';
import { ELEMENT_ID, IMAGE_SRC } from '../constants/variables';
import { storeAuthData } from '../actions';
import { fetchUser } from '../actionCreators';
import { getIsFetchingUser } from '../reducers/rootReducer';
import { APP_NAME } from '../constants/strings';
import { detectIE } from '../utils';

const propTypes = {
  storeAuthData: PropTypes.func.isRequired,
  fetchUser: PropTypes.func.isRequired,
  isFetchingUser: PropTypes.bool.isRequired,
};

export class Login extends Component {
  // Sets up localstorage listener for cross-tab communication
  // since the authentication requires the user to be redirected
  // to another page and then redirected back to a return URL with the token.
  componentDidMount() {
    window.addEventListener('storage', this.storageEventListener);
  }

  componentWillUnmount() {
    window.removeEventListener('storage', this.storageEventListener);
  }

  storageEventListener = (event) => {
    const { storeAuthData, fetchUser } = this.props;
    const authData = JSON.parse(localStorage.getItem(event.key));

    // store the auth data in Redux store
    storeAuthData(authData);

    fetchUser();
  }

  openNewTab = link => window.open(link, '_black')
  onLoginBtnClick = () => this.openNewTab(SSO_LOGIN_ENDPOINT)
  onIdirLoginBtnClick = () => this.openNewTab(SSO_IDIR_LOGIN_ENDPOINT)
  onBceidLoginBtnClick = () => this.openNewTab(SSO_BCEID_LOGIN_ENDPOINT)

  render() {
    const { isFetchingUser } = this.props;
    const isIE = detectIE();

    return (
      <section className="login">
        {isIE &&
          <article className="login__no-support-browser">
            <div className="login__no-support-browser__title">
              Your internet browser is not supported.
            </div>
            <div>
              <div>
                Please visit {APP_NAME} using a supported browser:
                <a href="https://www.google.com/chrome" target="_blank" rel="noopener noreferrer"> Google Chrome,</a>
                <a href="https://www.mozilla.org/en-US/firefox/new" target="_blank" rel="noopener noreferrer"> Firefox,</a>
                <a href="https://support.apple.com/en-ca/safari" target="_blank" rel="noopener noreferrer"> Safari </a>
                (Mac only).
              </div>
              If you choose to continue with this browser the application may not work as intended.
            </div>
          </article>
        }
        <article className="login__header">
          <img className="login__header__logo" src={IMAGE_SRC.LOGIN_LOGO} alt="Logo" />
          <div className="login__header__title">{APP_NAME}</div>
        </article>
        <article className="login__paragraph1">
          <Segment basic>
            <Loading active={isFetchingUser} />
            <div className="login__signin__container">
              <div className="login__signin__title">Sign In</div>
              <div className="login__signin__text1">to continue to {APP_NAME}</div>
              <div className="login__signin__text2">We use the BCeID for authentication.</div>
              <a
                className="login__signin__text3"
                href="https://portal.nrs.gov.bc.ca/web/client/bceid"
                target="_blank"
                rel="noopener noreferrer"
              >
                Learn more about BCeID here.
              </a>
              <Button
                id={ELEMENT_ID.LOGIN_BCEID_BUTTON}
                className="login__signin__button"
                primary
                fluid
                style={{ height: '50px' }}
                onClick={this.onBceidLoginBtnClick}
              >
                Login as Agreement Holder
              </Button>
              <div className="login__signin__link-container">
                <div
                  role="button"
                  tabIndex="0"
                  onClick={this.onIdirLoginBtnClick}
                >
                  Range Staff Login
                </div>
                <div className="login__line" />
                <div
                  role="button"
                  tabIndex="0"
                  onClick={this.onLoginBtnClick}
                >
                  Admin Login
                </div>
              </div>
            </div>
          </Segment>
        </article>
        <article className="login__paragraph2">
          <div className="login__paragraph2__title">What is {APP_NAME}?</div>
          <div className="login__paragraph2__text">
            We are making it easier for you to submit and amend Range Use Plans. This new service will enable you to submit your plan electronically, save drafts and print versions.
            Your Range Officer is also getting new tools to allow so that together we can manage the land with greater accuracy.
          </div>
        </article>
        <article className="login__paragraph3">
          <div className="container">
            <div className="login__paragraph4__content">
              <div className="login__paragraph-cell">
                <hr className="login__title-spacer" />
                <div className="login__paragraph3__title">
                  New simplified Range Use Plan across BC.
                </div>
                <div className="login__paragraph3__text">
                  There is now a common Range Use Plan across BC.
                  This common plan will enable greater efficiencies in service delivery.
                </div>
              </div>
              <div className="login__paragraph-cell">
                <img
                  className="login__paragraph3__image"
                  src={IMAGE_SRC.LOGIN_PARAGRAPH3}
                  alt="paragraph3_image"
                />
              </div>
            </div>
          </div>
        </article>
        <article className="login__paragraph4">
          <div className="container">
            <div className="login__paragraph4__content">
              <div className="login__paragraph-cell">
                <img
                  className="login__paragraph4__image"
                  src={IMAGE_SRC.LOGIN_PARAGRAPH4}
                  alt="paragraph4_image"
                />
              </div>
              <div className="login__paragraph-cell">
                <hr className="login__title-spacer" />
                <div className="login__paragraph4__title">
                  Submit your Range Use Plan faster then ever.
                </div>
                <div className="login__paragraph4__text">
                  No more mail or paper. When you submit a Range Use Plan for review or amendment your Range Staff Officer can review it and you will know its status.
                </div>
              </div>
            </div>
          </div>
        </article>
        <article className="login__paragraph5">
          <div className="container">
            <div className="login__paragraph5__content">
              <div className="login__paragraph-cell">
                <hr className="login__title-spacer" />
                <div className="login__paragraph5__title">
                  Easier login with BCeID
                </div>
                <div className="login__paragraph5__text">
                  With your BCeID you can view all of your Range Use Plans and amendments. We have included ways to save time in drafting your work
                </div>
              </div>
              <div className="login__paragraph-cell">
                <img
                  className="login__paragraph5__image"
                  src={IMAGE_SRC.LOGIN_PARAGRAPH5}
                  alt="paragraph5_image"
                />
              </div>
            </div>
          </div>
        </article>
        <article className="login__paragraph6">
          <div className="container">
            <div className="login__footer">
              Copyright
              <div className="login__line" />
              Accessability
              <div className="login__line" />
              Privacy
              <div className="login__line" />
              Disclaimer
            </div>
          </div>
        </article>
      </section>
    );
  }
}

const mapStateToProps = state => (
  {
    isFetchingUser: getIsFetchingUser(state),
  }
);

Login.propTypes = propTypes;
export default connect(mapStateToProps, { storeAuthData, fetchUser })(Login);
