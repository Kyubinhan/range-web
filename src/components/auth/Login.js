import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Button } from 'semantic-ui-react';

import { parseQuery } from '../../handlers';
import { SSO_AUTH_LOGIN_ENDPOINT } from '../../constants/api';
import { LOGIN_LOGO_SRC } from '../../constants/variables';
import { login } from '../../actions/authActions';

const propTypes = {
  login: PropTypes.func.isRequired,
  location: PropTypes.shape({}).isRequired,
  loginState: PropTypes.shape({}).isRequired,
};

export class Login extends Component {
  componentDidMount() {
    const { login, location } = this.props;

    // grab the code from the redirect url
    const { code } = parseQuery(location.search);

    if (code) {
      login(code);
    }
  }

  onLoginBtnClick = () => {
    window.open(SSO_AUTH_LOGIN_ENDPOINT, '_self');
  }

  render() {
    const {
      loginState,
    } = this.props;

    return (
      <div className="login">
        <img
          className="login__image"
          src={LOGIN_LOGO_SRC}
          alt="login-img"
        />

        <div className="login__title">
          My Range Application
        </div>

        <div className="login__button">
          <Button
            id="login-button"
            loading={loginState.isLoading || false}
            disabled={loginState.isLoading || false}
            primary
            fluid
            onClick={this.onLoginBtnClick}
          >
            Login
          </Button>

        </div>
        <a
          className="login__change-link"
          href="https://summer.gov.bc.ca"
          target="_blank"
          rel="noopener noreferrer"
        >
          Is your password expired?
        </a>
      </div>
    );
  }
}

const mapStateToProps = state => (
  {
    loginState: state.auth,
  }
);

Login.propTypes = propTypes;
export default connect(mapStateToProps, { login })(Login);

