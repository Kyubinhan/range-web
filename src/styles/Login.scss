.login {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100vh;

  &__title {
    
  }

  &__email {

  }

  &__password {

  }
}