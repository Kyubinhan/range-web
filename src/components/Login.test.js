import React from 'react';
import { shallow } from 'enzyme';
import { Login } from './Login';

const props = {};
const setupProps = () => {
  props.user = {
    id: 'id'
  };
  props.login = jest.fn();
};

beforeEach(() => {
  setupProps();
});

describe('Login', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Login {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('initializes with the correct state', () => {
    const wrapper = shallow(<Login {...props} />);
    expect(wrapper.state()).toEqual({
      email: '',
      password: ''
    });
  });

  describe('Event handlers', () => {
    it('`handleInput` set states with given id and value', () => {
      const wrapper = shallow(<Login {...props} />);
      const mockId = 'email';
      const mockValue = 'value';
      wrapper.instance().handleInput({ target: { id: mockId, value: mockValue }});
      expect(wrapper.state()[mockId]).toEqual(mockValue);
    });

    // TODO test onSubmit
  });
});