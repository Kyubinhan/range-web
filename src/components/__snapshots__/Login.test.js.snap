// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    login={[Function]}
    user={
        Object {
            "id": "id",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="login__title"
>
          MYRA APP
</div>,
        <div>
          Login
</div>,
        <div
          className="login__email"
>
          <label>
                    Email
          </label>
          <input
                    id="email"
                    onChange={[Function]}
                    placeholder="Enter Email"
                    type="text"
                    value=""
          />
</div>,
        <div
          className="login__password"
>
          <label>
                    Password
          </label>
          <input
                    id="password"
                    onChange={[Function]}
                    onKeyPress={undefined}
                    placeholder="Enter Password"
                    type="password"
                    value=""
          />
</div>,
        <button
          onClick={[Function]}
>
          Login
</button>,
      ],
      "className": "login",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "MYRA APP",
          "className": "login__title",
        },
        "ref": null,
        "rendered": "MYRA APP",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Login",
        },
        "ref": null,
        "rendered": "Login",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              Email
</label>,
            <input
              id="email"
              onChange={[Function]}
              placeholder="Enter Email"
              type="text"
              value=""
/>,
          ],
          "className": "login__email",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Email",
            },
            "ref": null,
            "rendered": "Email",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "email",
              "onChange": [Function],
              "placeholder": "Enter Email",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              Password
</label>,
            <input
              id="password"
              onChange={[Function]}
              onKeyPress={undefined}
              placeholder="Enter Password"
              type="password"
              value=""
/>,
          ],
          "className": "login__password",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Password",
            },
            "ref": null,
            "rendered": "Password",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "password",
              "onChange": [Function],
              "onKeyPress": undefined,
              "placeholder": "Enter Password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Login",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Login",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="login__title"
>
            MYRA APP
</div>,
          <div>
            Login
</div>,
          <div
            className="login__email"
>
            <label>
                        Email
            </label>
            <input
                        id="email"
                        onChange={[Function]}
                        placeholder="Enter Email"
                        type="text"
                        value=""
            />
</div>,
          <div
            className="login__password"
>
            <label>
                        Password
            </label>
            <input
                        id="password"
                        onChange={[Function]}
                        onKeyPress={undefined}
                        placeholder="Enter Password"
                        type="password"
                        value=""
            />
</div>,
          <button
            onClick={[Function]}
>
            Login
</button>,
        ],
        "className": "login",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "MYRA APP",
            "className": "login__title",
          },
          "ref": null,
          "rendered": "MYRA APP",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Login",
          },
          "ref": null,
          "rendered": "Login",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                Email
</label>,
              <input
                id="email"
                onChange={[Function]}
                placeholder="Enter Email"
                type="text"
                value=""
/>,
            ],
            "className": "login__email",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Email",
              },
              "ref": null,
              "rendered": "Email",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "email",
                "onChange": [Function],
                "placeholder": "Enter Email",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                Password
</label>,
              <input
                id="password"
                onChange={[Function]}
                onKeyPress={undefined}
                placeholder="Enter Password"
                type="password"
                value=""
/>,
            ],
            "className": "login__password",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Password",
              },
              "ref": null,
              "rendered": "Password",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "password",
                "onChange": [Function],
                "onKeyPress": undefined,
                "placeholder": "Enter Password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Login",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Login",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
